source/common/http/conn_pool_grid.cc

```diff
@@ -27,11 +27,16 @@ absl::string_view describePool(const ConnectionPool::Instance& pool) {
 static constexpr uint32_t kDefaultTimeoutMs = 300;
 
 std::string getSni(const Network::TransportSocketOptionsConstSharedPtr& options,
-                   Network::UpstreamTransportSocketFactory& transport_socket_factory) {
+                   Upstream::HostConstSharedPtr& host) {
   if (options && options->serverNameOverride().has_value()) {
     return options->serverNameOverride().value();
   }
-  return std::string(transport_socket_factory.defaultServerNameIndication());
+  std::string transport = std::string(host->transportSocketFactory().defaultServerNameIndication());
```

Comment:
nit: should this variable perhaps be named `default_sni`?

Reply:
good call